### ISSUE TITLE
Pressing Ctrl (when CapsLock is remapped to Ctrl) causes viewport to scroll to bottom

### ISSUE THREAD

**Issue by schweikert** [wez/wezterm#3306]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Mutter

### WezTerm version

20230320-124340-559cb7b0

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

No, and I'll explain why below

### Describe the bug

If I scrollback on a viewport and press Ctrl, for example to copy something, the viewport is scrolled to the bottom, as if a non-modifier key was pressed. If I set scroll_to_bottom_on_input to false, this doesn't happen.

This behavior is not new to this version. With the previous version it was the same.

### To Reproduce

- Scroll back
- Press Ctrl, for example Ctrl-Shift C to copy text

Result: the viewport is scrolled to bottom.

### Configuration

no config (I tested with wezterm -n)

### Expected Behavior

The viewport doesn't move when I press Ctrl.

**Comment by wez** [comment]
That doesn't happen for me!

Please run through https://wezfurlong.org/wezterm/troubleshooting.html#debugging-keyboard-related-issues
and share what is logged when you press the Ctrl key.

**Comment by schweikert** [comment]
I tested with the nightly version wezterm 20230321-091724-0b83c974 and unfortunately it doesn't seem to be fixed. Pressing Caps Lock (mapped to Ctrl) still causes the viewport to scroll to bottom.

**Comment by schweikert** [comment]
I use Gnome Tweaks to do the Caps Lock to Ctrl mapping and I noticed that if I use this layout option: *Make Caps Lock an additional Ctrl*, then Caps Lock behaves as Ctrl but wezterm does scroll-to-bottom on key press.

On the other hand, if I set the other option *Ctrl position -> Caps Lock as Ctrl*, the effect is the same except for wezterm that doesn't scroll to bottom. Looking at the debug log, the latter option changes the KeyEvent that wezterm sees:

```
14:21:42.677  DEBUG  wezterm_gui::termwindow                > KeyEvent(KeyEvent { key: Control, modifiers: NONE, repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(CapsLock), modifiers: NONE, phys_code: Some(CapsLock), raw_code: 66, repeat_count: 1, key_is_do
```

I will now use the "Ctrl position: Caps Lock as Ctrl" instead of the other option, but it probably would still be good to fix this to avoid others running into this problem.
